### DISCUSSION TITLE
Bulk push to pushgateway

### DISCUSSION THREAD

**Discussion by niteshsin786** [prometheus/client_python#746] [Q&A] [answered]
HI,
Does the library provide anything that can replace the below shell?

I want to push bulk data, but can you help me figure out how it's done with python?
==
cat <<EOF | curl -v --data-binary @- http://localhost:9091/metrics/job/pushgateway
test{id="11137241"} 3.20
test{id="13010551"} 6.03
test{id="11244135"} 5.30
EOF
==

**Comment by csmarchbanks** [comment, marked as answer]
You can see the example in the README for pushing to a gateway: https://github.com/prometheus/client_python#exporting-to-a-pushgateway.

In your case the code would look something like:
```python
from prometheus_client import CollectorRegistry, Gauge, push_to_gateway

registry = CollectorRegistry()
g = Gauge('test', 'test metric', ['id'], registry=registry)
g.labels("11137241").set(3.2)
...
push_to_gateway('localhost:9091', job='pushgateway', registry=registry)
```
